BREAKING.md

```diff
@@ -29,6 +30,29 @@ The `sharingLinkToRedeem` property is removed from the `IOdspResolvedUrl` interf
 ### Replace `createCreateNewRequest` function with `createOdspCreateContainerRequest` function
 The `createCreateNewRequest()` is removed and replaced with `createOdspCreateContainerRequest()` in the `odsp-driver` package. If any instances of `createCreateNewRequest()` are used, replace them with `createOdspCreateContainerRequest()` by importing it from `@fluidframework/odsp-driver` package.
 
+### Deprecate IFluidObject and introduce FluidObject
+This release deprecates the interface `IFluidObject` and introduces the utility type [`FluidObject`](https://github.com/microsoft/FluidFramework/blob/main/common/lib/core-interfaces/src/provider.ts). The primary reason for this change is that the module augmentation used by `IFluidObject` creates excessive type coupling where a small breaking change in any type exposed off `IFluidObject` can lead to type error in all usages of `IFluidObject`.
+On investigation we also found that the uber type `IFluidObject` wasn't genenerally necessary, as consumers generally only used a small number of specific types that they knew in advance.
+
+Given these points, we've introduced [`FluidObject`](https://github.com/microsoft/FluidFramework/blob/main/common/lib/core-interfaces/src/provider.ts). `FluidObject` is a utility type that is used in both it's generic and non-generic forms.
+
+The non-generic `FluidObject` is returned or taken in cases where the specific functionally isn't known, or is different based on scenario. You'll see this usage for things like `scope` and the request pattern.
+
+The non-generic `FluidObject` is a hint that the generic form of `FluidObject` should be used to inspect it. For example
+``` typescript
+    const maybeView: FluidObject<IFluidHTMLView> = requestFluidObject(container, "/");
```

Comment:
Subjective nit:  I don't love calling these `maybeFoo`.  Other ideas include `mayProvideFoo` or just `fluidObject` or `viewFluidObject`.

Reply:
My preference is to only use "maybe" in the Haskell sense, i.e. it's either the thing or it's nothing (`Type | undefined`, not `TypeA | TypeB`).  So e.g. when getting from a map, searching a list, etc.